conan/tools/cmake/toolchain/blocks.py

```diff
@@ -559,14 +561,16 @@ def context(self):
             cppinfo = req.cpp_info.aggregated_components()
             build_build_paths.extend(cppinfo.builddirs)
             build_bin_paths.extend(cppinfo.bindirs)
+        has_build_context = hasattr(self._conanfile, "settings_build") # always true in Conan 2.0
+        bin_paths = build_bin_paths if has_build_context else host_bin_paths
```

Comment:
Is `self._conanfile.dependencies.build` always empty with 1 profile? Even when there are build requirements?